packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -213,18 +211,19 @@ const serviceHeaderDefaults = {
 export class EngineReportingAgent<TContext = any> {
   private options: EngineReportingOptions<TContext>;
   private apiKey: string;
-  private report!: FullTracesReport;
-  private reportSize!: number;
+  private reports: { [schemaHash: string]: FullTracesReport } = Object.create(
+    null,
+  );
+  private reportSizes: { [schemaHash: string]: number } = Object.create(null);
   private reportTimer: any; // timer typing is weird and node-specific
   private sendReportsImmediately?: boolean;
   private stopped: boolean = false;
-  private reportHeader: ReportHeader;
+  private reportHeaders: { [schemaHash: string]: ReportHeader } = Object.create(
```

Comment:
I'm not saying you have to go back and change this (but if you're motivated to, do!), but I would have suggested a single map from schemaHash to an object with 3 fields rather than 3 parallel maps where you gotta just hope the keys all stay the same.